lib/bootsnap/load_path_cache/cache.rb

```diff
@@ -44,14 +44,20 @@ def load_dir(dir)
 
       # Try to resolve this feature to an absolute path without traversing the
       # loadpath.
-      def find(feature)
+      def find(feature, try_extensions: true)
         reinitialize if (@has_relative_paths && dir_changed?) || stale?
         feature = feature.to_s.freeze
+
         return feature if absolute_path?(feature)
-        return expand_path(feature) if feature.start_with?('./')
+
+        if feature.start_with?('./')
+          return try_extensions ? expand_path(feature) : File.expand_path(feature).freeze
+        end
+
         @mutex.synchronize do
-          x = search_index(feature)
+          x = search_index(feature, try_extensions: try_extensions)
           return x if x
+          return x unless try_extensions
```

Comment:
just `return unless try_extensions`. If you get there it means `x = nil`, so no point retuning `x`.

Reply:
oh, right :woman_facepalming: